pkg/cmd/browse/browse.go

```diff
@@ -128,15 +128,8 @@ func runBrowse(opts *BrowseOptions) error {
 	}
 
 	if opts.CommitFlag {
-		if opts.LastCommit == nil {
-			httpClient, err := opts.HttpClient()
-			if err != nil {
-				return err
-			}
-			opts.LastCommit = func() (*git.Commit, error) { return api.LastCommit(api.NewClientFromHTTP(httpClient), baseRepo) }
-		}
-		commit, err := opts.LastCommit()
-		if err == nil {
+		commit, err := opts.GitClient.LastCommit()
+		if err == nil && commit != nil {
```

Comment:
1. If there was an error, that error should be surfaced to the user and the command aborted. Is there a reason to intentionally silence it?
2. What are the cases when `commit == nil`?

Reply:
Oops, commit should never be nil. 

I was only silencing the error because that was how the code was before.

https://github.com/cli/cli/commit/5bdaab882b77e2c870a50d208284dc5c374108c1

I think it might be silenced because the git error wouldn't be understandable, and it might make more sense to open up the default branch if there hasn't been a commit pushed yet?
```
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
/usr/bin/git: exit status 128
```

Reply:
5dbaab8